backend/src/Altinn.Authorizationadmin/Altinn.AuthorizationAdmin.Persistance/DelegationMetadataRepository.cs

```diff
@@ -76,15 +77,24 @@ public async Task<DelegationChange> InsertDelegation(DelegationChange delegation
         }
 
         /// <inheritdoc/>
-        public async Task<DelegationChange> GetCurrentDelegationChange(string altinnAppId, int offeredByPartyId, int? coveredByPartyId, int? coveredByUserId)
+        public async Task<DelegationChange> GetCurrentDelegationChange(string? altinnAppId, string? resourceRegistryId, int offeredByPartyId, int? coveredByPartyId, int? coveredByUserId)
         {
             try
             {
                 await using NpgsqlConnection conn = new NpgsqlConnection(_connectionString);
                 await conn.OpenAsync();
-
-                NpgsqlCommand pgcom = new NpgsqlCommand(getCurrentDelegationChangeSql, conn);
-                pgcom.Parameters.AddWithValue("_altinnAppId", altinnAppId);
+                NpgsqlCommand pgcom = new NpgsqlCommand();
+                if (!string.IsNullOrWhiteSpace(resourceRegistryId))
+                {
+                    pgcom = new NpgsqlCommand(getCurrentDelegationChangeBasedOnResourceRegistryIdSql, conn);
+                    pgcom.Parameters.AddWithValue("_resourceregistryid", resourceRegistryId);
+                }
+                else if (!string.IsNullOrWhiteSpace(altinnAppId) || altinnAppId != "/")
```

Comment:
if the altinnAppId is ever "/" then there is an earlier bug in the code which still tries to build an appId without Org and App being the resource identifiers. So why is this needed?